libs/ledger/src/lib/ledger-export-form.tsx

```diff
@@ -0,0 +1,173 @@
+  const [dateTo, setDateTo] = useState('');
+  const maxFromDate = formatForInput(new Date(dateTo || now.current));
+  const maxToDate = formatForInput(now.current);
+  const { data } = useAssetsDataProvider();
+  const assets = (data || []).reduce((aggr, item) => {
+    aggr[item.id] = item.symbol;
+    return aggr;
+  }, {} as Record<string, string>);
+  const [isDownloading, setIsDownloading] = useState(false);
+  const [assetId, setAssetId] = useState(Object.keys(assets)[0]);
+  useEffect(() => {
+    if (!assetId) {
+      setAssetId(Object.keys(assets)[0]);
+    }
+  }, [assetId, assets]);
+  const VEGA_URL = useEnvironment((store) => store.VEGA_URL);
+  const protohost = VEGA_URL ? getProtoHost(VEGA_URL) : '';
+  const disabled = Boolean(!assetId || isDownloading);
+  const assetDropDown = (
+    <DropdownMenu
+      trigger={
+        <DropdownMenuTrigger
+          disabled={isDownloading}
+          id="asset"
+          className="block w-full"
+        >
+          {assets[assetId]}
+        </DropdownMenuTrigger>
+      }
+    >
+      <DropdownMenuContent>
+        {Object.keys(assets).map((assetKey) => (
+          <DropdownMenuItem
+            key={assetKey}
+            onSelect={() => setAssetId(assetKey)}
+          >
+            {assets[assetKey]}
+          </DropdownMenuItem>
+        ))}
+      </DropdownMenuContent>
+    </DropdownMenu>
+  );
+
+  const link = useMemo(() => {
+    if (protohost && partyId && assetId) {
+      const defaultFrom = subDays(now.current, 7);
+      const dateToUrl = dateTo
+        ? `&dateRange.endTimestamp=${toNanoSeconds(dateTo)}`
+        : '';
+      const dateFromUrl = dateFrom
+        ? `&dateRange.startTimestamp=${toNanoSeconds(dateFrom)}`
+        : dateTo
+        ? ''
+        : `&dateRange.startTimestamp=${toNanoSeconds(defaultFrom)}`;
+      return `${protohost}/api/v2/ledgerentry/export?partyId=${partyId}&assetId=${assetId}${dateFromUrl}${dateToUrl}`;
+    }
+    return '';
+  }, [protohost, partyId, assetId, dateFrom, dateTo]);
+  const startDownload = useCallback(() => {
+    setIsDownloading(true);
+    fetch(link)
+      .then(async (resp) => {
+        const { headers } = resp;
+        const nameHeader = headers.get('content-disposition');
+        const filename =
+          nameHeader?.split('=').pop() ?? DEFAULT_EXPORT_FILE_NAME;
+        const blob = await resp.blob();
+        if (blob) {
+          const link = document.createElement('a');
+          link.href = URL.createObjectURL(blob);
+          link.download = filename;
+          link.click();
+        }
+      })
+      .catch((err) => {
+        localLoggerFactory({ application: 'ledger' }).error(
+          'Download file',
+          err
+        );
+      })
+      .finally(() => {
+        setIsDownloading(false);
+      });
+  }, [link]);
+
+  if (!protohost || Object.keys(assets).length === 0) {
+    return null;
+  }
+  return (
+    <div className="w-full h-full flex justify-center">
```

Comment:
This should use a form with an onSubmit handler

Reply:
done